common/autoware_overlay_rviz_plugin/autoware_minimap_overlay_rviz_plugin/src/path_overlay.cpp

```diff
@@ -0,0 +1,179 @@
+// Copyright 2024 The Autoware Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#include "include/path_overlay.hpp"
+
+#include <QDebug>
+
+#include <iomanip>
+#include <sstream>
+
+PathOverlay::PathOverlay() : vehicle_lat_(0.0), vehicle_lon_(0.0)
+{
+}
+
+void PathOverlay::setVehiclePosition(double lat, double lon)
+{
+  vehicle_lat_ = lat;
+  vehicle_lon_ = lon;
+}
+
+void PathOverlay::setPathPoints(
+  const std::vector<PathPoint> & path_points, double origin_lat, double origin_lon)
+{
+  path_points_geo_.clear();
+  for (const auto & point : path_points) {
+    auto geo_point = localToGeographic(point.x, point.y, origin_lat, origin_lon);
+    path_points_geo_.emplace_back(geo_point);
+  }
+}
+
+void PathOverlay::setProjectionInfo(
+  const std::string & projector_type, const std::string & mgrs_grid)
+{
+  projector_type_ = projector_type;
+  mgrs_grid_ = mgrs_grid;
+}
+
+std::pair<double, double> PathOverlay::localToGeographic(
+  double local_x, double local_y, double origin_lat, double origin_lon)
+{
+  if (projector_type_ == "MGRS") {
+    return localToGeographicMGRS(local_x, local_y, origin_lat, origin_lon, mgrs_grid_);
+  } else {
+    return localToGeographicUTM(local_x, local_y, origin_lat, origin_lon);
+  }
```

Comment:
It's probably better to use the [variables](https://github.com/tier4/tier4_autoware_msgs/blob/a6d8211121615793fb8fdbe3a04a4acea33a09bb/tier4_map_msgs/msg/MapProjectorInfo.msg#L5) defined in the message instead of raw text.

```suggestion
  if (projector_type_ == tier4_map_msgs::msg::MapProjectorInfo::MGRS) {
    return localToGeographicMGRS(local_x, local_y, origin_lat, origin_lon, mgrs_grid_);
  } else if (projector_type_ == tier4_map_msgs::msg::MapProjectorInfo::LOCAL_CARTESIAN_UTM) {
    return localToGeographicUTM(local_x, local_y, origin_lat, origin_lon);
  }
```